operator/docs/enhancements/short_lived_tokens_authentication.md

```diff
@@ -0,0 +1,355 @@
+- The LokiStack administrator is not required to expose credentials/identity-tokens/etc. to the cloud provider when operating in that workflow.
+
+### Non-Goals
+
+- Provide auxiliary and comprehensive tooling to automate preparing the cloud provider IAM resources for short-lived token authentication.
+
+## Proposal
+
+The following proposal describes the required changes and validations for each object storage secret for each major cloud provider (Azure, AWS, GCP).
+
+### API Extensions
+
+The present proposal introduces a new field in the `ObjectStorageSecretSpec` namely `CredentialsMode`:
+
+```go
+/ CredentialMode represents the type of authentication used for accessing the object storage.
+//
+// +kubebuilder:validation:Enum=static;token;managed
+type CredentialMode string
+
+const (
+    // CredentialModeStatic represents the usage of static, long-lived credentials stored in a Secret.
+    // This is the default authentication mode and available for all supported object storage types.
+    CredentialModeStatic CredentialMode = "static"
+    // CredentialModeToken represents the usage of short-lived tokens retrieved from a credential source.
+    // In this mode the static configuration does not contain credentials needed for the object storage.
+    // Instead, they are generated during runtime using a service, which allows for shorter-lived credentials and
+    // much more granular control. This authentication mode is not supported for all object storage types.
+    CredentialModeToken CredentialMode = "token"
+    // CredentialModeManaged represents the usage of short-lived tokens retrieved from a credential source.
+    // This mode is similar to CredentialModeToken,but instead of having a user-configured credential source,
+    // it is configured by the environment, for example the Cloud Credential Operator in OpenShift.
+    // This mode is only supported for certain object storage types in certain runtime environments.
+    CredentialModeManaged CredentialMode = "managed"
+)
+
+// ObjectStorageSecretSpec is a secret reference containing name only, no namespace.
+type ObjectStorageSecretSpec struct {
+...
+    // CredentialMode can be used to set the desired credential mode for authenticating with the object storage.
+    // If this is not set, then the operator tries to infer the credential mode from the provided secret and its
+    // own configuration.
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    CredentialMode CredentialMode `json:"credentialMode,omitempty"`
+}
+```
+
+The purpose of the `CredentialMode` is to override the detected credentials type of object storage secrects, e.g. in environments like STS-managed OpenShift clusters, where the operator is automatically setup to use `managed` but the user might want to use `static` to storage logs for example to Minio on the same cluster instead to AWS S3.
```

Comment:
Do we need to mention, that setting `credentialMode` is optional more explicitly?